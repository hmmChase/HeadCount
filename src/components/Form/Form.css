input {
  border: 5px solid white;
  box-shadow: inset 0 0 8px rgba(64, 98, 135, 0.34),
    0 0 16px rgba(64, 98, 135, 0.23);
  padding: 10px;
  background: rgba(255, 255, 255, 0.5);
  margin: 0 0 10px 0;
  font-size: 2.5vh;
  margin: auto;
  width: 50%;
  font-family: 'Crimson Text', serif;
}

.formContainer {
  width: 100%;
  background-color: #1d060614;
  box-shadow: 2px 2px 2px gainsboro;
  height: 26vh;
  padding-top: 3vh;
  animation-duration: 5s;
  animation-name: header-slide-open;
}

@keyframes header-slide-open {
  from {margin-top: -200px};
  to {margin-top: 0}

}

h1 {
  color: #416387;
  font-size: 6vh;
  text-shadow: 2px 2px white;
  margin-top: 0;
}

input:focus {
  outline: none;
}
