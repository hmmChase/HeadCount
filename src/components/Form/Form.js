import React, { Component } from 'react';

class Form extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchInput: ''
    }

    this.handleInputChange = this.handleInputChange.bind(this)
  }

  handleInputChange (event) {
    let inputvalue = event.target.value
    this.setState ({
      searchInput: inputvalue
    })
    this.props.filterDistricts(inputvalue)
  }

  render () {
    return (
      <form>
        <input
          type='text'
          name='district'
          placeholder='Search Districts'
          value={this.state.searchInput}
          onChange={(e) => this.handleInputChange(e)}
        />
      </form>
    )
  }
}

export default Form