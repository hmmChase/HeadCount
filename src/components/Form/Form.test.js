import React from 'react';
import Form from '../Form/Form'
const mockProps =  {
  filterDistricts: jest.fn() 
}

describe('Form', () => {
  let form
  beforeEach(() => {
    form = shallow(<Form {...mockProps} />)
  })
  
  it('matches snapshot', () => {
    expect(form).toMatchSnapshot()
  })

  describe('handleInputChange', () => {
    it('updates state when input is changes', () => {
      let input = form.find('input')

      input.simulate('change', {target : { value : 'hello'}})

      expect(form.state().searchInput).toEqual('hello')
    })

    it('should call filterDistricts on change', () => {
      const mockEvent = { target: { value: 'hello' } }
      form.instance().handleInputChange(mockEvent)

      expect(mockProps.filterDistricts).toHaveBeenCalled()
    })
  })
})

import { shallow } from 'enzyme'