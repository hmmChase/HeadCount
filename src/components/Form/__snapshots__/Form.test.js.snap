// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    filterDistricts={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Welcome To Headcount 2.0
</h1>,
        <form>
          <input
                    name="district"
                    onChange={[Function]}
                    placeholder="Search Districts"
                    type="text"
                    value=""
          />
</form>,
      ],
      "className": "formContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome To Headcount 2.0",
        },
        "ref": null,
        "rendered": "Welcome To Headcount 2.0",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="district"
            onChange={[Function]}
            placeholder="Search Districts"
            type="text"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "district",
            "onChange": [Function],
            "placeholder": "Search Districts",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome To Headcount 2.0
</h1>,
          <form>
            <input
                        name="district"
                        onChange={[Function]}
                        placeholder="Search Districts"
                        type="text"
                        value=""
            />
</form>,
        ],
        "className": "formContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome To Headcount 2.0",
          },
          "ref": null,
          "rendered": "Welcome To Headcount 2.0",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="district"
              onChange={[Function]}
              placeholder="Search Districts"
              type="text"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "district",
              "onChange": [Function],
              "placeholder": "Search Districts",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
